pkg/kafka/policy.go

```diff
@@ -281,9 +222,42 @@ func (req *RequestMessage) ruleMatches(rule api.PortRuleKafka) bool {
 // rules. The function will return true if the policy allows the message,
 // otherwise false is returned.
 func (req *RequestMessage) MatchesRule(rules []api.PortRuleKafka) bool {
+	reqTopicsMap := map[string]bool{}
+
+	// Maintain a map of all topics in the request
+	// Only if all the topics in a request are
+	// allowed by the rules, we should allow a request.
+	topics := req.GetTopics()
+	for _, topic := range topics {
+		reqTopicsMap[topic] = true
+	}
+
 	for _, rule := range rules {
-		if req.ruleMatches(rule) {
-			return true
+		switch len(topics) {
+		case 0:
+			if req.ruleMatches(rule, "") {
+				return true
+			}
+		default:
+			if rule.Topic != "" {
+				_, ok := reqTopicsMap[rule.Topic]
```

Comment:
Either benefit from your use of `true` values in the map:
```
if reqTopicsMap[rule.Topic] && req.ruleMatches(rule, rule.Topic) {
...
```

Or decide to continue ignoring the map's values but then define it properly as a `map[string]struct{}` instead of a `map[string]bool`.

Reply:
Should I not worry about if the entry does not exist? i.e check `ok`?

Reply:
All I meant is that `reqTopicsMap[rule.Topic]` returns the value for that key if it's found (and that value will always be `true` in this method), or the default value for the type (`false` for type `bool`) when it's not found. You should take advantage of that.

Reply:
done